backend/model_server/encoders.py

```diff
@@ -89,8 +89,18 @@ def _embed_openai(
         # OpenAI does not seem to provide truncation option, however
         # the context lengths used by Danswer currently are smaller than the max token length
         # for OpenAI embeddings so it's not a big deal
-        response = self.client.embeddings.create(input=texts, model=model)
-        return [embedding.embedding for embedding in response.data]
+        try:
+            response = self.client.embeddings.create(input=texts, model=model)
+            return [embedding.embedding for embedding in response.data]
+        except BaseException as e:
```

Comment:
Please be thorough with your changes. It's very easy to get the full context of if this makes sense.

<img width="480" alt="image" src="https://github.com/user-attachments/assets/3b2a72dd-60d0-4f33-bb83-621e3974dcc8">